x-pack/solutions/search/plugins/enterprise_search/server/config.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { PluginConfigDescriptor } from '@kbn/core/server';
+
+export const configSchema = schema.object({
+  enabled: schema.boolean({ defaultValue: true }),
+  hasConnectors: schema.boolean({ defaultValue: true }),
+  hasDefaultIngestPipeline: schema.boolean({ defaultValue: true }),
+  hasDocumentLevelSecurityEnabled: schema.boolean({ defaultValue: true }),
+  hasIncrementalSyncEnabled: schema.boolean({ defaultValue: true }),
+  hasNativeConnectors: schema.boolean({ defaultValue: true }),
+  hasWebCrawler: schema.boolean({ defaultValue: false }),
+  ui: schema.object({
+    enabled: schema.boolean({ defaultValue: true }),
+  }),
+});
+
+export type ConfigType = TypeOf<typeof configSchema>;
+
+export const config: PluginConfigDescriptor<ConfigType> = {
+  deprecations: ({ unusedFromRoot }) => [
+    unusedFromRoot('enterpriseSearch.host', { level: 'critical' }),
```

Comment:
Using `unusedFromRoot` here because the next thing we want to do is rename this plugin's config path from `enterpriseSearch` to `xpack.search`. 

I'm not 100% sure if we want these deprecations here with v9 breaking change, but I am being cautious including them. I can remove them if we think they can be removed. cc: @seanstory @markjhoy

Reply:
Wouldn't them not work at all anyways ? If they don't do anything, they can be gone IMO. Better Sean and Mark comment though.

Reply:
AFAIK, they need to be there for 8.x at least for the Kibana Upgrade Assistant... for 9.x and beyond, since Enterprise Search won't be there, so in `main` and 9.x they don't need to exist...

Reply:
I'll remove these and create a ticket to add config deprecations for all of these to 8.x branch